forc-plugins/forc-doc/src/render/link.rs

```diff
@@ -23,9 +23,28 @@ pub(crate) struct DocLinks {
 }
 impl Renderable for DocLinks {
     fn render(self, _render_plan: RenderPlan) -> Result<Box<dyn RenderBox>> {
+        let mut links_vec = Vec::new();
+        // sort the doc links alphabetically
+        // for the AllDoc page, sort based on the module prefix
+        match self.style {
+            DocStyle::AllDoc(_) => {
+                for (block_title, mut doc_link) in self.links {
+                    doc_link.sort_by(|a, b| {
+                        a.module_info.module_prefixes[1].cmp(&b.module_info.module_prefixes[1])
+                    });
+                    links_vec.push((block_title, doc_link));
+                }
+            }
```

Comment:
I tested locally and I think there may a be a problem here. It seems that it only does a shallow alphabetization for the all doc items so you end up with things like `alloc::realloc` coming before `alloc::alloc_bytes`. Not a big deal really, but would you mind making a follow up issue/PR for it?

Also thank you for tackling this!

Reply:
Was there a reason just using `.sort()` didn't work? Just curious